state/cloudcredentials.go

```diff
@@ -299,6 +284,26 @@ func validateCloudCredentials(
 	return ops, nil
 }
 
+func validateCredentialForCloud(nuage cloud.Cloud, tag names.CloudCredentialTag, credential Credential) error {
+	if tag.Cloud().Id() != nuage.Name {
+		return errors.NotValidf("cloud %q", tag.Cloud().Id())
```

Comment:
perhaps this error message should include both cloud names - the one from the cloud and the one from the credential so the nature of the mismatched is surfaced

Reply:
Yes \o/ I agree - am thrilled we think the same..
I wanted to make sure that we have both. This PR includes the error message that we'd get in this case. It'd be something along the lines of:
```
 'validating credential "stratus/bob/foobar" for cloud "dummy": cloud "stratus" not valid`
```